src/app/server/Server.cpp

```diff
@@ -222,6 +222,8 @@ CHIP_ERROR Server::Init(const ServerInitParams & initParams)
 
     mReportScheduler = initParams.reportScheduler;
 
+    mTermsAndConditionsProvider = initParams.termsAndConditionsProvider;
```

Comment:
This should be conditional T&Cs enabled, otherwise we have to pay for the provider flash space even if not used.

Reply:
The way it was implemented should only take up an extra few bytes. But I moved it to a TermsAndConditionsManager::GetInstance(). Let me know if this is better or worse.

Reply:
It is far worse to have a singleton. The issue w/ not having the `#if` is that the full dependency tree of the code of the feature (and its processing) is paid by all devices, not just the ones that support ESF.

The goal is to separate building and inclusion of the dependencies to avoid ESF anything showing-up in a non-ESF-using product.

Reply:
See the latest code. I think you jumped the gun on the review yesterday. It is now ready...